.github/workflows/publish.yml

```diff
@@ -18,6 +28,7 @@ jobs:
 
   publish-tag:
     if: startsWith(github.ref, 'refs/tags/')
+    needs: version-check
```

Comment:
Here a presence of the `if` clause means that the job will run disregarding whether the `version-check` succeeds if I understood the [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) correctly.

> If a job fails or is skipped, all jobs that need it are skipped **unless the jobs use a conditional expression that causes the job to continue.**

Reply:
It is not: the documentation describes the situation when `needs` overrided with `if always`, see this example run below

https://github.com/DifferentialOrange/queue/actions/runs/4471886246

Reply:
It seems, it is spcifically about `if: ${{ always() }}`. @DifferentialOrange shows me an example and it actually works (failed version check discards the following jobs).